tests/test_filesystem.py

```diff
@@ -140,3 +140,103 @@ def test_put():
 
         assert os.path.exists(remote_file)
         assert os.stat(remote_file).st_mode & 0o777 == 0o755
+
+
+def test_move():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file = f"{temp_dir}/script.sh"
+        with open(file, "wb") as f:
+            lines = ("#! /bin/bash\n"
+                     "echo 'Hello world'\n")
+            f.write(lines.encode())
+        os.chmod(file, 0o755)
+
+        fs, _ = filesystem.resolve_filesystem_and_path(file)
+
+        remote_temp_file = f"{temp_dir}/copied_script.sh"
+        fs.put(file, remote_temp_file)
+
+        remote_file = f"{temp_dir}/moved_script.sh"
+        fs.move(remote_temp_file, remote_file)
+
+        assert os.path.exists(remote_file)
+        assert os.stat(remote_file).st_mode & 0o777 == 0o755
+
+
+@pytest.mark.skip
+def test_put_hdfs():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file = f"{temp_dir}/script.sh"
+        with open(file, "wb") as f:
+            lines = ("#! /bin/bash\n"
+                     "echo 'Hello world'\n")
+            f.write(lines.encode())
+        os.chmod(file, 0o755)
+
+        user = os.environ.get("USER")
+        hdfs_file = f"viewfs://root/user/{user}/{temp_dir}/script.sh"
+
+        fs, _ = filesystem.resolve_filesystem_and_path(hdfs_file)
+
+        remote_file = f"{temp_dir}/copied_script.sh"
+        fs.put(file, remote_file)
+
+        assert fs.exists(remote_file)
+        assert fs.st_mode(remote_file) & 0o777 == 0o755
+
+
+@pytest.mark.skip
+def test_move_hdfs():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file = f"{temp_dir}/script.sh"
+        with open(file, "wb") as f:
+            lines = ("#! /bin/bash\n"
+                     "echo 'Hello world'\n")
+            f.write(lines.encode())
+        os.chmod(file, 0o755)
+
+        user = os.environ.get("USER")
+        hdfs_file = f"viewfs://root/user/{user}/{temp_dir}/script.sh"
+
+        fs, _ = filesystem.resolve_filesystem_and_path(hdfs_file)
+
+        remote_temp_file = f"{temp_dir}/copied_script.sh"
+        fs.put(file, remote_temp_file)
+
+        remote_file = f"{temp_dir}/moved_script.sh"
+        fs.move(remote_temp_file, remote_file)
+
+        assert fs.exists(remote_file)
+        assert fs.st_mode(remote_file) & 0o777 == 0o755
+
+
+@pytest.mark.skip

```

Comment:
Did not read at all the review yet but it seems you forgot to not skip these tests

Reply:
I'm not sure the github CI supports the HDFS tests.

Reply:
Then it means tests are not properly written or mocked, they should not really call HDFS otherwise it is considered as IT